### ISSUE TITLE
Bookmarks becomes Observing List

### ISSUE THREAD

**Issue by Jocelyn1109** [Stellarium/stellarium#976]
Hi,
I create a ticket to formalize the development of this new feature. 
This ticket will be attached to the pull request.
The bookmarks will be completely redone to become observation lists.

Here the requirment :

 The Bookmarks tool will be deep refactored and visually splitted on 2 window - the first windows for using lists of bookmarked objects and the second list for adding/editing lists of objects. S
[Doc1.pdf](https://github.com/Stellarium/stellarium/files/4213417/Doc1.pdf)
[Doc1.pdf](https://github.com/Stellarium/stellarium/files/4213423/Doc1.pdf)
[Doc2.pdf](https://github.com/Stellarium/stellarium/files/4213424/Doc2.pdf)

ee sketches in the attachment.

Some comments:

- 1) We may store all lists in one file - for example observingLists.json - and this file just storing list of lists;  

- 2) Any list should storing 2 additional hidden properties for manage import/export of the data - unique ID and data of latest changes. Unique ID already implemented for current bookmarks.

- 3) Export and import tool should export/import list of bookmarks as currently implemented (subset of observingLists.json file)

- 4) All list shouldn't be stored translated data - english only names of designations. Translation should be applied on stage of visualization (creating table of view) data. Formal we may store only English name (or designation) and coordinates of object (currently in bookmarks we use coordinates to create a custom marker for object, who is not finded in Stellarium's objects) - and adding the coordinates (for finded objects - current coordinates), magnitude and abbreviation of constellation into table on stage of visualization of list.

- 5) "Empty" observingLists.json should store one list without objects - Bookmarks.

- 6) All files in UTF-8 and we may use special extension for separate lists (an exchange format of data) - .sol (stellarium observing list) or .skylist, or something else.

- 7) In future we may add tool for import observing lists from stellarium.org website (online import data) - of course if this tool will be very appropriated by users and we'll get feedback.

- 8) At the moment we store the location and date of observing data for each bookmark item and this is not very frequently use by users. But if we'll feedback we may add date and location data for all items of list as property for list: adding 2 fields into window 2 for define location and date/time and 2 checkboxes in first window for enable/disable applies those data.

- 9) Selecting object should be implemented by double click on list of object (+maybe adding separate button for go to object as currently implemented in Bookmarks tool) 

- 10) Maybe adding column "Type" (Type of object) will be useful. In any way we should use model to manage the data (like in Bookmarks at the moment).

I will update regulary this ticket to show development progress.
Currently the developpment is on going.

Jocelyn
[Doc1.pdf](https://github.com/Stellarium/stellarium/files/4213410/Doc1.pdf)
[Doc2.pdf](https://github.com/Stellarium/stellarium/files/4213412/Doc2.pdf)

**Comment by martinber** [comment]
Hi,

Some kind of observability column would be useful, so the user can determine if the object is visible from the selected location.

As an example, months ago with some friends we tried to make an [observing list web application](https://dso.mbernardi.com.ar/). In this case, we added a column with a "visibility plot" (something like a monthly elevation AstroCalc graph)

I guess it's not an easy task, I just wanted to give you the idea just in case.

**Comment by axd1967** [comment]
@Jocelyn1109 what is the need for the unique ID?

**Comment by alex-w** [comment]
> @Jocelyn1109 what is the need for the unique ID?

Unique ID per list allow update and exchange data between different copies of one list.

**Comment by Jocelyn1109** [comment]
Hi @alex-w, currently in the bookmarks json file we have a unique uuid for each object.
For the observing list we will have a unique uuid for each list, but should we keep the uuid for objects ?

Jocelyn

**Comment by ultrapre** [comment]
Hello! Which uuid way in Python below is true one to generate the sample in Stellarium?
uuid1(),uuid3(),uuid4(),uuid5()

**Comment by alex-w** [comment]
@Jocelyn1109 sorry for late answer.

> For the observing list we will have a unique uuid for each list, but should we keep the uuid for objects?

This is a good question. UUID in Bookmarks was introduced for allowing adding one object multiple times. In new tool the one object  may added into the list once only. So, no need UUID for objects.

**Comment by Jocelyn1109** [comment]
@alex-w ok Alex many thanks :)

**Comment by Jocelyn1109** [comment]
@alex-w Hi, I put in attachment an example of observing list file.
Is it ok for you ?
[bookmarks.txt](https://github.com/Stellarium/stellarium/files/4408713/bookmarks.txt)

**Comment by alex-w** [comment]
Hi!

вт, 31 мар. 2020 г. в 18:32, Jocelyn GIROD <notifications@github.com>:

> @alex-w <https://github.com/alex-w> Hi, I put in attachment an example of
> observing list file.
> Is it ok for you ?
> bookmarks.txt
> <https://github.com/Stellarium/stellarium/files/4408713/bookmarks.txt>
>
Please see attachments.

obslist.txt: This is JSON file, contains one observing list and it may be
used as format for exchange the data - export into one file, import and
share on stellarium.org for example.

obslists.txt: This is JSON file, contains 2 observing lists at the moment -
"Bookmarks" and "80 planetaries". The "80 planetaries" list have also
location of observer (coordinates and planet) and time.

Right now array of objects contains one field for each element -
designation, but in future we may expand it by storing coordinates, if we
allow storing objects without designations or index to allow hand-made
storing the list.

The English and translated name, coordinates, magnitude and constellation
we will get from the founded object on the step of list parsing for
visualize the table.

-- 
With best regards, Alexander
Phone: +7 903 957 3596
Skype: alex.v.wolf

{ "observingList" : { 
      "description" : "80 planetaries for small telescopes (100-130 mm)",
      "jd" : 2458777.074287,
      "location" : "N45.0, E38.57, Earth",
      "name" : "80 planetaries",
      "objects" : [ 
          { "designation" : "NGC 7662" },
          { "designation" : "NGC 7009" },
          { "designation" : "NGC 7293" },
          { "designation" : "NGC 6741" },
          { "designation" : "NGC 6751" },
          { "designation" : "NGC 6772" },
          { "designation" : "NGC 6778" },
          { "designation" : "NGC 6781" },
          { "designation" : "NGC 6790" },
          { "designation" : "NGC 6803" },
          { "designation" : "NGC 6804" },
          { "designation" : "NGC 6807" },
          { "designation" : "NGC 6852" },
          { "designation" : "IC 2149" },
          { "designation" : "IC 3568" },
          { "designation" : "NGC 1501" },
          { "designation" : "IC 2165" },
          { "designation" : "IC 1747" },
          { "designation" : "IC 289" },
          { "designation" : "NGC 40" },
          { "designation" : "NGC 7354" },
          { "designation" : "NGC 246" },
          { "designation" : "NGC 4361" },
          { "designation" : "NGC 6826" },
          { "designation" : "NGC 6833" },
          { "designation" : "NGC 6884" },
          { "designation" : "NGC 6894" },
          { "designation" : "NGC 7008" },
          { "designation" : "NGC 7026" },
          { "designation" : "NGC 7027" },
          { "designation" : "NGC 7048" },
          { "designation" : "PK 086-08.1" },
          { "designation" : "NGC 6891" },
          { "designation" : "NGC 6905" },
          { "designation" : "NGC 6543" },
          { "designation" : "NGC 1535" },
          { "designation" : "NGC 1360" },
          { "designation" : "PK 194+02.1" },
          { "designation" : "NGC 2371" },
          { "designation" : "NGC 2392" },
          { "designation" : "SH 2-274" },
          { "designation" : "NGC 6058" },
          { "designation" : "IC 4593" },
          { "designation" : "NGC 6210" },
          { "designation" : "NGC 2610" },
          { "designation" : "NGC 3242" },
          { "designation" : "IC 5217" },
          { "designation" : "PK 100-08.1" },
          { "designation" : "IC 418" },
          { "designation" : "PK 342+27.1" },
          { "designation" : "PK 064+15.1" },
          { "designation" : "M 57" },
          { "designation" : "NGC 2346" },
          { "designation" : "IC 4634" },
          { "designation" : "NGC 6309" },
          { "designation" : "NGC 6369" },
          { "designation" : "NGC 6572" },
          { "designation" : "PK 190-17.1" },
          { "designation" : "NGC 2022" },
          { "designation" : "M 76" },
          { "designation" : "IC 351" },
          { "designation" : "IC 2003" },
          { "designation" : "NGC 2348" },
          { "designation" : "NGC 2440" },
          { "designation" : "NGC 2452" },
          { "designation" : "NGC 2818" },
          { "designation" : "NGC 6886" },
          { "designation" : "NGC 6445" },
          { "designation" : "NGC 6563" },
          { "designation" : "NGC 6567" },
          { "designation" : "NGC 6629" },
          { "designation" : "NGC 6644" },
          { "designation" : "NGC 6818" },
          { "designation" : "NGC 6302" },
          { "designation" : "IC 1295" },
          { "designation" : "NGC 1514" },
          { "designation" : "M 97" },
          { "designation" : "NGC 3132" },
          { "designation" : "M 27" }
        ],
      "observingListID" : "{16e942bf-3cbd-4bd8-a9c6-7eac30e22f51}",
      "sorting" : "right ascension"
    } 
}

{ 
  "observingLists" : { 
      "{16e942bf-3cbd-4bd8-a9c6-7eac30e22f51}" : { 
          "description" : "80 planetaries for small telescopes (100-130 mm)",
          "jd" : 2458777.074287,
          "location" : "N45.0, E38.57, Earth",
          "name" : "80 planetaries",
          "objects" : [ 
              { "designation" : "NGC 7662" },
              { "designation" : "NGC 7009" },
              { "designation" : "NGC 7293" },
              { "designation" : "NGC 6741" },
              { "designation" : "NGC 6751" },
              { "designation" : "NGC 6772" },
              { "designation" : "NGC 6778" },
              { "designation" : "NGC 6781" },
              { "designation" : "NGC 6790" },
              { "designation" : "NGC 6803" },
              { "designation" : "NGC 6804" },
              { "designation" : "NGC 6807" },
              { "designation" : "NGC 6852" },
              { "designation" : "IC 2149" },
              { "designation" : "IC 3568" },
              { "designation" : "NGC 1501" },
              { "designation" : "IC 2165" },
              { "designation" : "IC 1747" },
              { "designation" : "IC 289" },
              { "designation" : "NGC 40" },
              { "designation" : "NGC 7354" },
              { "designation" : "NGC 246" },
              { "designation" : "NGC 4361" },
              { "designation" : "NGC 6826" },
              { "designation" : "NGC 6833" },
              { "designation" : "NGC 6884" },
              { "designation" : "NGC 6894" },
              { "designation" : "NGC 7008" },
              { "designation" : "NGC 7026" },
              { "designation" : "NGC 7027" },
              { "designation" : "NGC 7048" },
              { "designation" : "PK 086-08.1" },
              { "designation" : "NGC 6891" },
              { "designation" : "NGC 6905" },
              { "designation" : "NGC 6543" },
              { "designation" : "NGC 1535" },
              { "designation" : "NGC 1360" },
              { "designation" : "PK 194+02.1" },
              { "designation" : "NGC 2371" },
              { "designation" : "NGC 2392" },
              { "designation" : "SH 2-274" },
              { "designation" : "NGC 6058" },
              { "designation" : "IC 4593" },
              { "designation" : "NGC 6210" },
              { "designation" : "NGC 2610" },
              { "designation" : "NGC 3242" },
              { "designation" : "IC 5217" },
              { "designation" : "PK 100-08.1" },
              { "designation" : "IC 418" },
              { "designation" : "PK 342+27.1" },
              { "designation" : "PK 064+15.1" },
              { "designation" : "M 57" },
              { "designation" : "NGC 2346" },
              { "designation" : "IC 4634" },
              { "designation" : "NGC 6309" },
              { "designation" : "NGC 6369" },
              { "designation" : "NGC 6572" },
              { "designation" : "PK 190-17.1" },
              { "designation" : "NGC 2022" },
              { "designation" : "M 76" },
              { "designation" : "IC 351" },
              { "designation" : "IC 2003" },
              { "designation" : "NGC 2348" },
              { "designation" : "NGC 2440" },
              { "designation" : "NGC 2452" },
              { "designation" : "NGC 2818" },
              { "designation" : "NGC 6886" },
              { "designation" : "NGC 6445" },
              { "designation" : "NGC 6563" },
              { "designation" : "NGC 6567" },
              { "designation" : "NGC 6629" },
              { "designation" : "NGC 6644" },
              { "designation" : "NGC 6818" },
              { "designation" : "NGC 6302" },
              { "designation" : "IC 1295" },
              { "designation" : "NGC 1514" },
              { "designation" : "M 97" },
              { "designation" : "NGC 3132" },
              { "designation" : "M 27" }
            ],
          "sorting" : "right ascension"
        },
      "{306aeefa-45fc-4638-ac00-07b10e302ce4}" : { 
          "description" : "Bookmarks list",
          "name" : "Bookmarks",
          "objects" : [ 
              { "designation" : "Jupiter" },
              { "designation" : "HIP 81377" },
              { "designation" : "HIP 57632" },
              { "designation" : "M 1" },
              { "designation" : "Saturn" }
            ],
          "sorting" : "designation"
        }
    },
  "shortName" : "Observing lists for Stellarium",
  "version" : 1.0
}

**Comment by Jocelyn1109** [comment]
@alex-w Hi Alex. I don't realy understand. The file I put is the file generated by my code.
I must change a lot of things but no problem for that.
I want to be sure of the needed modifications before.
There are no RA, DEC, FOV, Magnitude and Constellation store in the json file ?
But we need to thoses informations for the fonction goToObject and those informations should appear in the dialog.

Thanks

Jocelyn

**Comment by alex-w** [comment]
> There are no RA, DEC, FOV, Magnitude and Constellation store in the json file ?

Yes, all those data we can get from founded object

> But we need to thoses informations for the fonction goToObject and those informations should appear in the dialog.

By the fact we need designation or name for method goToObject only. But in the future we may expand obsserving lists to objects without designations and in this case we will store coordinates into JSON. But this is on second step, right now we need solve problem for objects with designations - deep-sky objects, bright starts, Solar system objects.

**Comment by Jocelyn1109** [comment]
@alex-w Ok I will modify my code :)

**Comment by Jocelyn1109** [comment]
@alex-w Hello Alex the code is on going.
For the "version" information in the json file for the moment it is hard coded but in the futur where we will find this information ?

Thanks

Jocelyn

**Comment by alex-w** [comment]
It’s version of JSON format - in the future we may expand or change it and version info may be very helpful for create a “reading rules”. We may set version to 1.0 (default value) if the “version” variable is not existing in the file.

**Comment by Jocelyn1109** [comment]
@alex-w I put in attachment an example of the empty file.
The file will be generated the first time we open the observing list dialog if it doesn't exist.
The "default list" will be not loaded in the dialog and will be deleted when the first list is created.
[observingList-empty.txt](https://github.com/Stellarium/stellarium/files/4449335/observingList-empty.txt)


Are you agree ?

Jocelyn

**Comment by alex-w** [comment]
But see item V in the list of comments:
> "Empty" observingLists.json should store one list without objects - Bookmarks.

I think storing empty JD & location is not needed here

**Comment by Jocelyn1109** [comment]
Yes, I saw the information V...
I will delete JD and Location for the moment.

**Comment by alex-w** [comment]
I think you may store into JSON flag for list, who should be loaded by default

**Comment by Jocelyn1109** [comment]
I am not sure to understand. Do you want adding a flag in list information to say if this is the default list ?
What is the use of empty JSON file ?
For me we can only create this file when we create the first list (button new list).

**Comment by alex-w** [comment]
>I am not sure to understand. Do you want adding a flag in list information to say if this is the default list ?

Yes. Of course right now we have one empty list - Bookmarks, but in future the list of lists will be contains few items and we may propose to user choose the latest list when he open the GUI. How it do? We may use additional flag to implement the "chooser of list by default"

**Comment by Jocelyn1109** [comment]
Ok we could add a chexcbox in the dialog with label: "set list as default" and at the loading this list will be load :)

**Comment by alex-w** [comment]
Why just not remember latest selected list without extra checkbox? ;)

**Comment by Jocelyn1109** [comment]
I put a screen shot with the checkbox :)
![observingListDialog](https://user-images.githubusercontent.com/43777557/78797579-2c283f80-79b8-11ea-9ea2-a5046d67a472.png)

**Comment by Jocelyn1109** [comment]
Hi, just a little screenshot to see the progress.
As you can see the "highlight all" works with a big observing list and with good performance :)
![observingListDialog1](https://user-images.githubusercontent.com/43777557/78882304-2c781780-7a58-11ea-9d1d-a855bff24fff.png)

Jocelyn

**Comment by Jocelyn1109** [comment]
@alex-w @guillaumechereau 

Hi all, juste a little checkpoint to explain some behaviors I implemented when we open the observing list dialog.

**Case 1:** there is no json file (observingList.json) already created:
The dialog is open with the list(s) combo box empty and the "array" of objects empty.
Buttons _Highlight All_, _Clear Highlight(s)_ and _Edit list_ are disable. Only _New list_ button is enable because in this case we can only create a new list.

![obsList-nojson](https://user-images.githubusercontent.com/43777557/79224528-81060300-7e5b-11ea-9f56-3c3693ceceb4.png)


**Case 2:** there is a json file but no default list choosen by the user:
The dialog is open and the list of observing list names is loaded in the combo box. Array is empty and buttons _Highlight All_, _Clear Highlight(s)_ and _Edit list_ are disable. Only _New list_ button is enable.

![obsList-json-nodefault](https://user-images.githubusercontent.com/43777557/79224739-efe35c00-7e5b-11ea-9092-70afaad35bba.png)


**Case 3:** there is a json file with default list choosen by the user:
The dialog is open and the default list name is selected in the combo box. The list is loaded in the "array" of objects and the description too. All butttons are enable.

![obsList-json-default](https://user-images.githubusercontent.com/43777557/79224998-59636a80-7e5c-11ea-92a6-05d8533950fd.png)

![default-list-json](https://user-images.githubusercontent.com/43777557/79225523-21105c00-7e5d-11ea-9a7d-35b82dfd3226.png)


Additionnal informations: when there is no list selected, array and description are clear and _Highlight All_, _Clear Highlight(s)_ and _Edit list_ are disable.

Jocelyn

**Comment by Jocelyn1109** [comment]
Another screen shot !
![obsList-two-dialogs](https://user-images.githubusercontent.com/43777557/79434945-5dfc6000-7fcf-11ea-8508-09534f87b138.png)

**Comment by Jocelyn1109** [comment]
@alex-w @guillaumechereau 
In an observing list could we have a duplicate entry (see screen shot) or should i prevent it ?
![double](https://user-images.githubusercontent.com/43777557/114710507-14536c00-9d2e-11eb-8923-152a35301290.png)

Jocelyn

**Comment by gzotti** [comment]
Just at thought: support OpenAstronomyLog? This seems to be a somewhat accepted standard I saw in a discussion elsewhere.

**Comment by alex-w** [comment]
> @alex-w @guillaumechereau
> In an observing list could we have a duplicate entry (see screen shot) or should i prevent it ?

This is very good question. After some thinking for it I can suggest that on first step (this is our case) probably we shouldn't allow duplicates in the lists.

**Comment by alex-w** [comment]
> Just at thought: support OpenAstronomyLog? This seems to be a somewhat accepted standard I saw in a discussion elsewhere.

The OAL cannot be applicable here, because this is not an observations logger. You may use an idea, that we prepare a powerful edition of bookmarks here.

**Comment by Jocelyn1109** [comment]
@alex-w @gzotti @guillaumechereau 

Ok Alexander I will modify the code to not have duplicate object in observing list.
It is a good idea because in case of observing list, there is no sense in having the same object twice :)

Jocelyn

**Comment by alex-w** [comment]
This is not a "bookmarks with date" :) This is tool for creating target lists for observations (initial planning) or highlights (didactic purposes or various demos, like show all Messier objects). The current Bookmarks tool supported one list of objects and this is not very useful as show the practice. The Observing Lists tool expanding bookmarks for various lists of objects with quick switch between lists.

OAL is standardised structure for saving notes about observations (telescope, eyepiece, sky condition, target, notes, etc.) and it can be used together with planning of observations, but... this is separate big task.

**Comment by gzotti** [comment]
But "bookmark with date" is also useful, as shows my own practice. Please keep the optional date stamp available, and repeated observations of the same target. Imagine a list of observations of Jupiter's moons, or a list of Lunar eclipses. Always the same object! To extend, it should be possible to add observing location (and even landscape?) to the list.

**Comment by Jocelyn1109** [comment]
@alex-w @gzotti @guillaumechereau 
Currently the date as julian Day is recorded in the the observingList.json but not used (see scrennshot).
I think that for this first development we could prevent duplicate item in the list as requested by Alex.
We can always improve the Observing List in the futur ans I thinks that after several months of use by end-users there will be features to improve or add.

![bookmark-jd](https://user-images.githubusercontent.com/43777557/114830626-b3c73c00-9dcc-11eb-98ce-3f918d1b8287.png)

Jocelyn

**Comment by gzotti** [comment]
And what if I want to display/demonstrate/explain the same solar eclipse observed from different locations? Or display the parallax shift in Lunar position between two observer positions (to explain occultations)? This is possible with the current bookmarks which allows date and location per object entry. 

It should be possible to extend the "objects" entries with optional date and location tags. Only full duplicates may then be excluded, but {"designation": "moon"} and {"designation": "moon", "jd": "2451545.0" } are different entries. And by the way, a "jd" is a single double precision float, not a complex string with years,months, hours and minutes.
We have discussed "user object lists" for years. There was rumour of an "extension manager" that should allow this. But it never appeared.  

Planning usage scenarios in advance is better than designing *usage limitations* in a feature replacement and hoping for possible later expansion.

**Comment by Jocelyn1109** [comment]
@gzotti 
" And by the way, a "jd" is a single double precision float, not a complex string with years,months, hours and minutes."  it's corrected.

**Comment by Jocelyn1109** [comment]
@alex-w  @guillaumechereau 

Hi all, I did some modifications in Observing List dialog and features.

In the main dialog:
- I added the field "Creation date" which allow to display the creation date of the list. That means that, when you select an object in the list and double click on it, you will go to this object at the date creation (jd) of the list (if you didn't do that there is no sens to have observing list, because observing lists must make it possible to prepare for future observations).
For example: I want to create an  observation list with all objects of the Solar system in date of "2024-04-19 22:00:00". I change the date in Stellarium to "2024-04-19 22:00:00" and I create my liste. In fact as requested there no date per object but a global date for the list, and of course you can create several lists with another date.

- I added the button "Delete list" to delete the selected list.

![obslist_1](https://user-images.githubusercontent.com/43777557/115370940-d771f400-a1c9-11eb-8d36-5d1fb8f51129.png)

In the create/edit dialog:
- there is now two buttons to exit: one to exit with save and another to exit without saving.

![obslist_2](https://user-images.githubusercontent.com/43777557/115372857-9da1ed00-a1cb-11eb-8f7e-d8b1f8962e73.png)

Jocelyn

**Comment by gzotti** [comment]
That means I would need to create one 1-entry dated list per solar eclipse per place, instead of a list of views of the same solar eclipse from different sites on earth, or a list of a sequence of solar eclipses e.g. from the same Saros series, seen from different places?

Please don't miss the chance for a flexible, widely usable solution! What is being developed here is a different usage scenario than the current bookmarks. It looks like something that should go into an optional plugin. Please do not simply replace the current bookmarks by this solution.

**Comment by Jocelyn1109** [comment]
@gzotti  I think you can read a detailed specification of what you want and I will do.

**Comment by axd1967** [comment]
@Jocelyn1109 @gzotti @alex-w @Atque @treaves 

Introducing **lists** of bookmarks is a **great** improvement because users had to fiddle with bookmark files all the time. This new feature now allows to play with **sets** of bookmarks.

But I'm a bit worried about this change request _because it has been merged into `master` (d5d8d667a1893d08e8abee9aea6b8bc7f34ac105) with significant shortcomings_.

- In https://github.com/Stellarium/stellarium/pull/1826 I added observations that have been _partly_ addressed in https://github.com/Jocelyn1109/stellarium/commit/232a5415b836400f4382105457726a3bcb6abcf0 (thx @Jocelyn1109 , but please consider at least to include the [tab order changes](https://github.com/Stellarium/stellarium/pull/1826/files?file-filters%5B%5D=.ui#diff-43557ea2b2b275f2a614a6185942f259e2071a172a497bc60b76052e398f102cR335-R346) too)
- in https://github.com/Stellarium/stellarium/pull/1816#issuecomment-890382301 I added critical observations

IMHO the merge should not have been done, because the current implementation shows oddities in UI as well as in code and has not been thoroughly tested.

_Please forgive me because I really hate to write all this_, because software contributions are dearly needed, we need developers such as @Jocelyn1109 , but **IMHO** seen the upcoming release (about two weeks) the merge to `master` ought to be reverted, pending a fix of the various observations. In its current state, that merge represents a **regression**, with users will lose their bookmarks as its most important problem. This _will_ hurt users. And the choice of UI flow is strange, as well as the duplication of code beneath it.

I really hope all the issues will be resolved **before the next release** because IMHO the current state of the implementation is not in the best shape.

**Comment by gzotti** [comment]
I second that. I never exported/imported or sent around bookmark files, but I did store events like eclipses with their respective observing locations and dates. I have not yet found out what to do (but I stopped trying after my first loading location 0/0/0 given more important problems). Yes, for me the bookmarks have lost value.

**Comment by axd1967** [comment]
Although I'm in no way responsible for Stellarium, I **really** recommend to revert the change, **it's no big deal**. Otherwise I predict that many users will be upset by this feature.

Also, this will give more time to @Jocelyn1109 to deal with the remaining issues, and for more review time from other devs, as this is now becoming too critical: the remaining issues should have been dealt with.

Maybe an additional observation: with the release date approaching, merging new features might be halted, maybe bug fixing merged, and focus should now switch to translators so they have time to translate what still needs to be translated, e.d to a few days before the next release.

**Comment by axd1967** [comment]
Another indication that there is something slightly irregular is that this has been merged to `master` and the issue remains open. An issue should - almost "atomically" - be closed once the issue has been merged.

**Comment by axd1967** [comment]
**less than a week to go and nothing is happening here. this is becoming worrying.**

on a side note: there should be a more strict release schedule.

e.g

D-7: no more feature merging
D-3: no more bugfixing
D-2: no more translations
D-1: production and final testing
D: release

**Comment by axd1967** [comment]
>  I don't know whether the change should be reverted, but I hope the lost functionality as written several times will be added again by the end of the year.

To stay on the safe side, it feels better not to build on hope and "see what happens at the end of the year": that's what feature branches are for. Let's be honest: the current state of the implementation is not very acceptable and should never have been merged in the first place. And I'm sure the change should be reverted.

The merge contains changes that seem irrelevant to this issue but might be critical to other Stellarium areas, such as eg 
- https://github.com/Stellarium/stellarium/commit/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105#diff-078269f68279672ed83f88bda98f4c80161a5118388580841958f7b7c308661cR853
- https://github.com/Stellarium/stellarium/commit/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105#diff-038af28b4b6b86f26327cf3429b6ca2a234edd33b58a7c9e7c16d3e4d50ca7f9R322
- https://github.com/Stellarium/stellarium/commit/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105#diff-038af28b4b6b86f26327cf3429b6ca2a234edd33b58a7c9e7c16d3e4d50ca7f9R1068
- https://github.com/Stellarium/stellarium/commit/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105#diff-e5c347e12bcb591a8e2d0eddb723591fe743c158a2300afd0f65d6c2c2f07f9bR98-R99

I'm beginning to wonder whether this is not just an accidental, unwanted merge?

I did not check whether these changes were also re-applied elsewhere.

(On a side issue, this indicates another area where care should be taken to avoid committing unrelated changes. This is a life-long battle of every software engineer against trying to do too many things at the same time; understandable, and can only be solved through discipline (and reviews).)

Reverting the merge also justifies to reopen the prematurely closed PR https://github.com/Stellarium/stellarium/pull/1813. in fact, that PR should not have been closed before this feature was complete.

@gzotti @alex-w @Jocelyn1109 @Atque @treaves @xalioth please have a look at this.

**Comment by axd1967** [comment]
And maybe this should also never be merged: https://github.com/Stellarium/stellarium/commit/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105#diff-0cde85c820c4accefc9f187ebd9a172975adbdcf63155c6403199a8f1c33034fR1-R4

**Comment by Jocelyn1109** [comment]
Hi I was in vacation.
The new version of the bookmarks should not be delivered until it is corrected and finished. It is in progress.
I will focus on the first specification (cf: pdf file).
If you want changes, update the specification and I will take it into account, but it will take a little longer.

Many thanks.

Jocelyn

**Comment by axd1967** [comment]
@Jocelyn1109 great!

Make sure to have a look at this: https://github.com/Stellarium/stellarium/pull/1826#issuecomment-920697034

**Comment by Jocelyn1109** [comment]
Ok I will see :)

**Comment by axd1967** [comment]
check also this:

```
stellarium/src/gui/ObsListCreateEditDialog.cpp:307:4: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  307 |    if ( !Location.isEmpty() )
      |    ^~
```

**Comment by axd1967** [comment]
The comment of https://github.com/Stellarium/stellarium/issues/1884#issuecomment-923814939 is extremely worrying.

This is all a matter of reverting a single commit once it has been determined that there are major issues here.

**Comment by gzotti** [comment]
Be worried then.

**Comment by axd1967** [comment]
One big issue with this issue is that date/time and observer location are not available (at least not in the current state). The argument that these features are not used much is not based on any objective measurement.

**Comment by axd1967** [comment]
Also, this does not belong in the repository, but in local git configuration because Stellarium does not depend on kdev4:

https://github.com/Stellarium/stellarium/blob/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105/.gitignore#L74

This belongs in `$GIT_DIR/info/exclude` (which is not shared). For more info, see http://book.git-scm.com/docs/gitignore ; as an example, see 
> "so you can redirect the output into your project exclude file ... 
> That way, you don’t litter the project with .gitignore files. This is a good option if you’re the only Git user on a Subversion [or any other] team, and your teammates don’t want .gitignore files in the project.
 "
on http://book.git-scm.com/book/en/v2/Git-and-Other-Systems-Git-as-a-Client

**Comment by axd1967** [comment]
Here is an example of a sensible approach: https://github.com/Stellarium/stellarium/pull/1880#discussion_r713977291.

**Comment by axd1967** [comment]
Also to check: indentation should be respected, see e.g

https://github.com/Stellarium/stellarium/commit/d5d8d667a1893d08e8abee9aea6b8bc7f34ac105#diff-038af28b4b6b86f26327cf3429b6ca2a234edd33b58a7c9e7c16d3e4d50ca7f9R1066

**Comment by axd1967** [comment]
**This will also break the Astrocalc window.**

**Comment by axd1967** [comment]
WUT used to export bookmarks, now this will be broken too.

**Comment by Jocelyn1109** [comment]
@axd1967 it is useless to post a message every day. Group your remarks in a document, otherwise some will be forgotten and not taken into account.
It's complicated to be inundated with comments every day especially when the development is underway and not finalized.

Many thanks.

**Comment by axd1967** [comment]
I understand. But there is a flaw in that logic: if several people would attach comments, you would still have to do the management of the inputs.

(And it is so worrying that no-one else seems to bother to evaluate such a critical issue. @Atque  @treaves please help ?)

**Comment by alex-w** [comment]
@Jocelyn1109 after get some feedback from community I think we ready to write a second edition of specification for this tool and implement it :)

**Comment by BlackApron** [comment]
Has anyone noticed that Stellarium crashes if you want to import a json file with an observing list created in another computer? (That is an observing list, the new one, not a bookmarks file).

**Comment by scandari** [comment]
I want to report that my stellarium 0.21.3 crashes or does nothing when trying to import a json file with an observing list. Even when lists are created with the same computer (just creating a new one, exporting, deleting and importing does nothing).

**Comment by Jocelyn1109** [comment]
Hi @scandari thank you for your feedback the problem is known and I will correct it and the Observing List V2.

Jocelyn

**Comment by ilkant** [comment]
I encourage you to download bookmarks from a JSON file on your computer. So not only from the www page.

This would create a great opportunity to share with other people their own observation lists and others could upload them to their own Stellarium. And you could have multiple sets of observations in multiple bookmark files.

Also, I would like an easier way to create bookmarks. Directly to the bookmark window without the editing window. Or even directly from Stellarium's starry sky, for example with a combination of keyboard and mouse clicks. E.g. ALT and clicking the mouse button. It would make bookmarking items much easier.

You are coding excellent application!

**Comment by ilkant** [comment]
The bookmark could also contain field information. You could write a reference to the article in the space magazine that tells about that object.
